datafusion/core/src/execution/context.rs

```diff
@@ -1643,6 +1621,72 @@ impl SessionState {
         self
     }
 
+    /// Creates a [`LogicalPlan`] from the provided SQL string
+    ///
+    /// See [`SessionContext::sql`] for a higher-level interface that also handles DDL
+    pub async fn create_logical_plan(&self, sql: &str) -> Result<LogicalPlan> {
```

Comment:
This is specifically broken out into a separate function so that database's that want to not handle DDL, such as IOx, don't have to re-implement this functionality. FYI @alamb